irods/parallel.py

```diff
@@ -293,19 +307,27 @@ def _io_part (objHandle, range_, file_, opr_, mgr_, thread_debug_id = '', queueO
 
 
 def _io_multipart_threaded(operation_ , dataObj_and_IO, replica_token, hier_str, session, fname,
-                           total_size, num_threads = 0, **extra_options):
+                           total_size, num_threads, **extra_options):
     """Called by _io_main.
     Carve up (0,total_size) range into `num_threads` parts and initiate a transfer thread for each one."""
     (D, Io) = dataObj_and_IO
     Operation = Oper( operation_ )
 
-    if num_threads < 1:
-        num_threads = RECOMMENDED_NUM_THREADS_PER_TRANSFER
-    num_threads = max(1, min(multiprocessing.cpu_count(), num_threads))
+    def bytes_range_for_thread( i, num_threads, total_bytes,  chunk ):
+        begin_offs = i * chunk
+        if i + 1 < num_threads:
+            end_offs = (i + 1) * chunk
+        else:
+            end_offs = total_bytes
+        return six.moves.range(begin_offs, end_offs)
+
+    P = total_size // num_threads
```

Comment:
is there a word that better describes what `P` represents?

Reply:
yes, it's the multipart size, the #bytes transferred per thread.  Maybe I'll use `partial_size`;

Reply:
or `bytes_per_thread`

Reply:
even better!